www/docs/server/procedures.md

```diff
@@ -5,336 +5,88 @@ sidebar_label: Define Procedures
 
-You're able to chain multiple parsers in order to make reusable publicProcedures for different parts of your application.
+The below example takes a user input and [authorizes](https://en.wikipedia.org/wiki/Authorization) them like protective towns-people. This is obviously a contrived example for simplicity, and not an appropriate way to securely authorize an application user, so in practice you may want to use some combination of [Headers](/docs/client/headers), [Context](context), [Middleware](middlewares), and [Metadata](metadata), to [authenticate](https://en.wikipedia.org/wiki/Authentication) and authorize your users.
 
 ```ts twoslash
-// @filename: trpc.ts
-import { initTRPC } from '@trpc/server';
+// @target: esnext
+import { TRPCError, initTRPC } from '@trpc/server';
 import { z } from 'zod';
 
-const t = initTRPC.create();
-
-export const middleware = t.middleware;
-export const router = t.router;
-
-
-/**
- * Public, unprotected publicProcedure
- **/
-export const publicProcedure = t.procedure;
-
-
 // ---cut---
 
-// ------------------------------
-// @filename: roomProcedure.ts
-// ------------------------------
-import { publicProcedure } from './trpc';
-import { z } from 'zod';
-
-/**
- * Create reusable publicProcedure for a chat room
- */
-export const roomProcedure = publicProcedure.input(
-  z.object({
-    roomId: z.string(),
-  }),
-);
-
-// ------------------------------
-// @filename: _app.ts
-// ------------------------------
-import { router } from './trpc';
-import { roomProcedure } from './roomProcedure';
-import { z } from 'zod';
-
-
-const appRouter = router({
-  sendMessage: roomProcedure
-    // Add extra input validation for the `sendMessage`-procedure
-    .input(
-      z.object({
-        text: z.string(),
-      }),
-    )
-    .mutation(({ input }) => {
-      //         ^?
-      // [...]
-    }),
-});
-
-export type AppRouter = typeof appRouter;
-```
-
-## Multiple Procedures
+export const t = initTRPC.create();
 
-To add multiple publicProcedures, you define them as properties on the object passed to `t.router()`.
+export const publicProcedure = t.procedure
 
-```tsx
-import { initTRPC } from '@trpc/server';
+export const authorizedProcedure = publicProcedure
+  .input(z.object({ townName: z.string() }))
+  .use((opts) => {
+    if (opts.input.townName !== 'Pucklechurch') {
+      throw new TRPCError({
+        code: 'FORBIDDEN',
+        message: "We don't take kindly to out-of-town folk",
+      });
+    }
 
-export const t = initTRPC.create();
-
-const router = t.router;
-const publicProcedure = t.procedure;
+    return opts.next();
+  });
 
-export const appRouter = router({
-  hello: publicProcedure.query(() => {
+export const appRouter = t.router({
+  hello: authorizedProcedure.query((opts) => {
     return {
-      text: 'hello world',
+      greeting: `Hello my friend`,
     };
   }),
-  bye: publicProcedure.query(() => {
+  goodbye: authorizedProcedure.mutation(() => {
```

Comment:
same as above, i get that it's difficult to give code samples for mutations because they need to mutate _something_, but having a mutation that just returns a string feels weird